From 70f9025950b355b4688b84f7dd87787e164dc192 Mon Sep 17 00:00:00 2001
From: Brandon Hurst <brandon.hurst@analog.com>
Date: Tue, 17 Feb 2026 10:56:21 -0800
Subject: [PATCH 32/38] drivers: usb: udc: max32: Reduce spurious resume events

UDC driver was submitting spurious resume events
when receiving DPACT events while the device was not
suspended.

- Change udc_max32_event callback to only respond to
DPACT events when suspended

Signed-off-by: Brandon Hurst <brandon.hurst@analog.com>
---
 drivers/usb/udc/udc_max32.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/udc/udc_max32.c b/drivers/usb/udc/udc_max32.c
index 5c561540a79..b083f53314f 100644
--- a/drivers/usb/udc/udc_max32.c
+++ b/drivers/usb/udc/udc_max32.c
@@ -566,8 +566,11 @@ static int udc_max32_event_callback(maxusb_event_t event, void *cbdata)
 		break;
 	case MAXUSB_EVENT_DPACT:
 		LOG_DBG("DPACT event occurred");
-		udc_set_suspended(dev, false);
-		udc_submit_sof_event(dev);
+		/* Only submit resume if device is suspended */
+		if (udc_is_suspended(dev)) {
+			udc_set_suspended(dev, false);
+			udc_submit_event(dev, UDC_EVT_RESUME, 0);
+		}
 		break;
 	case MAXUSB_EVENT_BRST:
 		LOG_DBG("BRST event occurred");
-- 
2.47.1

